crates/astria-sequencer/src/app.rs

```diff
@@ -315,10 +319,42 @@ impl App {
             .try_into()
             .map_err(|_| anyhow!("chain IDs commitment must be 32 bytes"))?;
 
-        let signed_txs = self
-            .execute_block_data(txs)
+        let expected_txs_len = txs.len();
+
+        let block_data = BlockData {
+            txs: txs.into_iter().collect(),
+            misbehavior: process_proposal.misbehavior,
+            height: process_proposal.height,
+            time: process_proposal.time,
+            next_validators_hash: process_proposal.next_validators_hash,
+            proposer_address: process_proposal.proposer_address,
+        };
+
+        let txs = self
+            .pre_execute_transactions(block_data)
             .await
-            .context("transactions failed to decode and execute")?;
+            .context("failed to prepare for executing block")?;
+
+        // we don't care about the cometbft max_tx_bytes here, as cometbft would have
+        // rejected the proposal if it was too large.
+        // however, we should still validate the other constraints, namely
+        // the max sequenced data bytes.
+        let mut block_size_constraints = BlockSizeConstraints::new(usize::MAX)
+            .context("failed to create block size constraints")?;
+
+        let (signed_txs, txs_to_include) = self
+            .execute_transactions_before_finalization(txs, &mut block_size_constraints)
+            .await
+            .context("failed to execute transactions")?;
+
+        // all txs in the proposal should be deserializable and executable
+        // if any txs were not deserializeable or executable, they would not have been
+        // returned by `execute_block_data`, thus the length of `txs_to_include`
+        // will be shorter than that of `txs`.
+        ensure!(
+            txs_to_include.len() == expected_txs_len,
```

Comment:
Can we require that `execute_transactions_before_finalization` only return with success if all transactions are executable and deserializable instead of adding this extra check?

Reply:
`execute_transactions_before_finalization` is also called in `prepare_proposal`, so it can't error if not all txs are executable. it's simplest to have this check here i think